packages/vanilla-renderers/src/complex/categorization/CategorizationRenderer.tsx

```diff
@@ -22,88 +22,105 @@
   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
   THE SOFTWARE.
 */
-import React from 'react';
+import React, { useMemo, useState } from 'react';
 import type { Categorization, Category, LayoutProps } from '@jsonforms/core';
+import { isVisible } from '@jsonforms/core';
 import {
-  RendererComponent,
   TranslateProps,
+  useJsonForms,
   withJsonFormsLayoutProps,
   withTranslateProps,
 } from '@jsonforms/react';
 import { CategorizationList } from './CategorizationList';
 import { SingleCategory } from './SingleCategory';
-import { isCategorization } from './tester';
 import { withVanillaControlProps } from '../../util';
-import type { VanillaRendererProps } from '../../index';
+import type { AjvProps, VanillaRendererProps } from '../../util';
 
 export interface CategorizationState {
   selectedCategory: Category;
 }
 
-class CategorizationRenderer extends RendererComponent<
-  LayoutProps & VanillaRendererProps & TranslateProps,
-  CategorizationState
-> {
-  onCategorySelected = (category: Category) => () => {
-    return this.setState({ selectedCategory: category });
-  };
+interface CategorizationProps {
+  selected?: number;
+  onChange?(selected: number, prevSelected: number): void;
+}
 
-  /**
-   * @inheritDoc
-   */
-  render() {
-    const { uischema, visible, getStyleAsClassName, t } = this.props;
-    const categorization = uischema as Categorization;
-    const classNames = getStyleAsClassName('categorization');
-    const masterClassNames = getStyleAsClassName('categorization.master');
-    const detailClassNames = getStyleAsClassName('categorization.detail');
-    const selectedCategory = this.findCategory(categorization);
-    const subcategoriesClassName = getStyleAsClassName(
-      'category.subcategories'
-    );
-    const groupClassName = getStyleAsClassName('category.group');
+export const CategorizationRenderer = ({
+  data,
+  uischema,
+  schema,
+  path,
+  selected,
+  t,
+  visible,
+  getStyleAsClassName,
+  onChange,
+}: LayoutProps &
+  VanillaRendererProps &
+  TranslateProps &
+  CategorizationProps &
+  AjvProps) => {
+  const ajv = useJsonForms().core.ajv;
```

Comment:
You should not need this because `ajv` is already given as part of the props as indicated by the `AjvProps` type. To actually get this handed in, make sure to wrap the export at the end of this file with `withAjvProps`